confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/MacroConverter.java

```diff
@@ -33,6 +35,16 @@
 @Role
 public interface MacroConverter
 {
+    /**
+     * Initialize the macro.
+     * @param confluenceConverter the confluence converter in use
+     * @param context the confluence input context in use
+     */
+    default void initialize(ConfluenceConverter confluenceConverter, ConfluenceInputContext context)
```

Comment:
> I guess this is because MentionMacroConverter is a singleton component, and confluenceConverter is per lookup (?)

No. Being singleton just means that the ComponentManager always return the same instance. If you `@Inject` ConfluenceInputContext it cannot be null.